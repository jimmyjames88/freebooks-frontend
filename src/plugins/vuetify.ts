// Styles
import '@mdi/font/css/materialdesignicons.css'
import '@/styles/main.scss'

// Vuetify
import { ThemeDefinition, createVuetify, defaults as vuetifyDefaults } from 'vuetify'

const fbTheme: ThemeDefinition = {
  dark: false,
  colors: {
    background: '#EFEFEF',
    'on-background': '#ffffff',
    surface: '#ffffff',
    'on-surface': '#9A8F99',
    primary: '#FE6D73',
    'on-primary': '#ffffff',
    secondary: '#17C3B2',
    'on-secondary': '#ffffff',
    tertiary: '#56CBF9',
    'on-tertiary': '#ffffff',
    info: '#227C9D',
    'on-info': '#ffffff'
  }
}

export default createVuetify({
  theme: {
    defaultTheme: 'fbTheme',
    themes: { fbTheme },
    icons: {
      defaultSet: 'mdi'
    }
  }
})
