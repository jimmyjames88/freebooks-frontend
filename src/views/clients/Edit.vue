<script lang="ts">
import { defineComponent } from 'vue';
import { Button, GradientContainer, TextField } from '@/components'
import API from '@/api'

export default defineComponent({
  name: 'Clients/Edit',
  components: { Button, GradientContainer, TextField },
  data: () => ({
    step: 1,
    _id: '',
    name: '',
    email: '',
    phone: '',
    website: '',
    address: {
      line1: '',
      line2: '',
      city: '',
      province: '',
      postal: '',
      country: ''
    }
  }),
  async mounted() {
    // load data from clients.show endpoint
    const response = await API.clients.edit(this.$route.params.clientId)
    // set data properties
    const { _id, name, email, phone, website, address } = response.data
    this._id = _id
    this.name = name
    this.email = email
    this.phone = phone
    this.website = website
    this.address.line1 = address.line1
    this.address.line2 = address.line2
    this.address.city = address.city
    this.address.province = address.province
    this.address.postal = address.postal
    this.address.country = address.country
  },
  methods: {
    save() {
      const { _id, name, email, phone, website, address } = this
      API.clients.update({  _id, name, email, phone, website, address }).then(() => {
        this.$router.push({ name: 'Clients/Show', params: { clientId: this._id } })
      }).catch((err: AxiosError) => {
        console.warn(err)
      })
    },
    cancel() {
      this.$router.go(-1)
    }
  }
})
</script>

<template>
  <GradientContainer>
    <template #left>
      <h1>Edit Client</h1>
    </template>
    <template #right>
      <v-form @submit.prevent="save">
        <v-window v-model="step"
          height="auto"
          direction="vertical"
          hide-delimiters
          :show-arrows="false">
          <v-window-item :value="1">
            <div>
              <TextField name="name" label="Name" v-model="name" />
              <TextField name="email" label="Email" v-model="email" />
              <TextField name="phone" label="Phone" v-model="phone" />
              <TextField name="website" label="Website" v-model="website" placeholder="https://" />
              <Button @click="cancel" color="transparent">
                Cancel
              </Button>
              <Button @click="step++" color="secondary">
                Next
              </Button>
            </div>
          </v-window-item>
          <v-window-item :value="2">
            <div>
              <TextField name="line1" v-model="address.line1" label="Line 1" />
              <TextField name="line2" v-model="address.line2" label="Line 2" />
              <TextField name="city" v-model="address.city" label="City" />
              <TextField name="province" v-model="address.province" label="Province" />
              <TextField name="postal" v-model="address.postal" label="Postal Code" />
              <TextField name="country" v-model="address.country" label="Country" />
              <Button color="transparent" @click="step--">Previous</Button>
              <Button type="submit" color="secondary">Save</Button>
            </div>
          </v-window-item>
        </v-window>
      </v-form>
    </template>
  </GradientContainer>
</template>