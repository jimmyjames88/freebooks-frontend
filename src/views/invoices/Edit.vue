<script lang="ts">
import { defineComponent } from 'vue'
import { useToast } from 'vue-toastification';
import { _Invoice, _LineItem } from '@jimmyjames88/freebooks-types'
import API from '@/api'
import InvoiceForm from './_Form.vue'
import { Header, Spinner } from '@/components'


export default defineComponent({
  name: 'Invoices/Edit',
  components: { Header, InvoiceForm, Spinner },
  data: (): {
    formData: {
      id: number | undefined,
      clientId: number | undefined,
      refNo: string,
      issueDate: Date,
      dueDate: Date,
      lineItems: _LineItem[],
      notes: string
    } | undefined,
    loading: boolean
  } => ({
    formData: undefined,
    loading: true
  }),
  mounted() {
    this.loadInvoice()
  },
  methods: {
    async loadInvoice() {
      try {
        const response = await API.invoices.show(this.$route.params.invoiceId)
        const { id, clientId, refNo, issueDate, dueDate, lineItems, notes } = response.data
        this.formData = {
          id,
          clientId,
          refNo,
          issueDate,
          dueDate,
          lineItems,
          notes
        }
      } catch(e) {
        console.error(e)
      } finally {
        this.loading = false
      }
    },
    async submitForm(data: Partial<_Invoice>) {
      try {
        this.loading = true
        await API.invoices.update({ 
          id: this.formData?.id,
          ...data 
        })
        useToast().success('Invoice saved')
        this.$router.push({ name: 'Invoices/Show', params: { invoiceId: data.id }})
      } catch(e) {
        console.error(e)
      } finally {
        this.loading = false
      }
    }
  }
})
</script>

<template>
  <Spinner v-if="loading" />
  <div v-show="!loading">
    <Header title="Edit Invoice" />
    <v-container>
      <InvoiceForm
        :formData="formData"
        :loading="loading"
        @submitForm="submitForm"
      />
    </v-container>
  </div>
</template>