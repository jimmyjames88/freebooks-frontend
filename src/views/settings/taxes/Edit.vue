<script lang="ts">
import { defineComponent } from 'vue'
import { AxiosError, AxiosResponse } from 'axios'
import { useToast } from 'vue-toastification'
import { _Tax } from '@jimmyjames88/freebooks-types'
import { GradientContainer } from '@/components'
import TaxesForm from './_Form.vue'
import API from '@/api'

export default defineComponent({
  name: 'Settings/Taxes/Create',
  components: { GradientContainer, TaxesForm },
  data: (): {
    loading: boolean,
    formData: Partial<_Tax>
  } => ({
    loading: true,
    formData: {
      name: '',
      rate: 0,
      type: undefined,
      default: false
    }
  }),
  mounted() {
    this.loadTax()
  },
  computed: {
    taxId(): number { return Number(this.$route.params.taxId) }
  },
  methods: {
    loadTax() {
      this.loading = true
      API.taxes.show(this.taxId).then((response: AxiosResponse) => {
        this.formData = response.data
      })
      .finally(() => {
        this.loading = false
      })
    },
    save(tax: _Tax) {
      API.taxes.update({ ...tax, id: this.taxId }).then((response: AxiosResponse) => {
        if (response.status === 200) {
          useToast().success('Changes saved')
          this.$router.push({ name: 'Settings' })
          return
        }
        throw new AxiosError('Error saving')
      }).catch((err: AxiosError) => {
        useToast().error(err.message)
      })
    },
    cancel() {
      this.$router.go(-1)
    }
  }
})
</script>

<template>
  <GradientContainer bgColor="#679dbd">
    <template #left>
      <h1>Edit Tax</h1>
    </template>
    <template #right>
      <TaxesForm v-if="!loading" @submitForm="save" @cancel="cancel" :formData="formData" />
    </template>
  </GradientContainer>
</template>

<style lang="scss" scoped>
// Todo Checkbox component
:deep(.v-label) {
  opacity: 1 !important;
}
</style>