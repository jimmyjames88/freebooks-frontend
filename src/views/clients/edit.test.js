import { nextTick } from 'vue'
import { mount } from '@vue/test-utils'
import { createVuetify } from "vuetify";
import { vi } from 'vitest'
import Edit from './Edit.vue'
import API from '@/api'

vi.mock('axios')

const mockClientResponse = {
  _id: 1,
  name: 'Test Name',
  email: 'test@example.com',
  phone: '1234567890',
  website: 'https://example.com',
  address: {
    line1: 'Address Line 1',
    line2: 'Address Line 2',
    city: 'Calgary',
    state: 'Alberta',
    postal: 'T1T1T1',
    country: 'Canada'
  }
}

vi.spyOn(API.clients, 'index').mockResolvedValue({ data: mockClientResponse })
vi.spyOn(API.clients, 'update').mockResolvedValue({ data: mockClientResponse })

describe('Edit.vue', () => {
  const vuetify = createVuetify()
  const $route = { params: { clientId: 1 }}
  let wrapper
  const $router = {
    go: vi.fn()
  }

  beforeEach(() => { 
    wrapper = mount(Edit, {
      global: {
        plugins: [vuetify],
        mocks: { $route, $router }
      },
    })

  })

  it('Renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  
  it('form submits the correct data', async () => {
    const formData = {
      name: 'Test Name',
      email: 'test@example.com',
      phone: '1234567890',
      website: 'https://example.com',
      line1: 'Address Line 1',
      line2: 'Address Line 2',
      city: 'Calgary',
      state: 'Alberta',
      postal: 'T1T1T1',
      country: 'Canada'
    }

    // fill out form
    const name = wrapper.find('input[name=name]').setValue(formData.name)
    const email = wrapper.find('input[name=email]').setValue(formData.email)
    const phone = wrapper.find('input[name=phone]').setValue(formData.phone)
    const website = wrapper.find('input[name=website]').setValue(formData.website)

    // click next button
    const next = wrapper.findAll('button').at(1).trigger('click')
    await nextTick()

    const line1 = wrapper.find('input[name=line1]').setValue(formData.line1)
    const line2 = wrapper.find('input[name=line2]').setValue(formData.line2)
    const city = wrapper.find('input[name=city]').setValue(formData.city)
    const state = wrapper.find('input[name=state]').setValue(formData.state)
    const postal = wrapper.find('input[name=postal]').setValue(formData.postal)
    const country = wrapper.find('input[name=country]').setValue(formData.country)
  
    // submit form
    wrapper.find('form').trigger('submit')

    // verify form submission and data
    expect(wrapper.emitted().submit).toBeTruthy()
  })

  it('goes back to last page when cancel button is clicked', async () => {
    const spy = vi.spyOn($router, 'go')

    // click cancel button
    const cancel = wrapper.findAll('button').at(0).trigger('click')
    await nextTick()
    
    expect(spy).toHaveBeenCalled()
  })

  it('goes to next step when next button is clicked', async () => {
    // click next button
    const next = wrapper.findAll('button').at(1).trigger('click')
    await nextTick()

    expect(wrapper.vm.step).toBe(2)
  })

  it('goes to previous step when previous button is clicked', async () => {
    // start on second step
    await wrapper.setData({ step: 2 })

    // click previous button
    const previous = wrapper.findAll('button').at(2).trigger('click')
    await nextTick()
    
    expect(wrapper.vm.step).toBe(1)
  })
})
