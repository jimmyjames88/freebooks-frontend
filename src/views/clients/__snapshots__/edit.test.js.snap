// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Edit.vue > Renders properly 1`] = `
"<div data-v-7a2e2f9a=\\"\\" class=\\"gradient-container\\">
  <div data-v-7a2e2f9a=\\"\\" class=\\"content\\">
    <div data-v-7a2e2f9a=\\"\\" class=\\"left\\">
      <h1>Edit Client</h1>
    </div>
    <div data-v-7a2e2f9a=\\"\\" class=\\"right\\">
      <form class=\\"v-form\\" novalidate=\\"\\">
        <div class=\\"v-window v-theme--light\\" height=\\"auto\\" hide-delimiters=\\"\\">
          <div class=\\"v-window__container\\">
            <transition-stub appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" disabled=\\"true\\">
              <div class=\\"v-window-item v-window-item--active\\">
                <div>
                  <div data-v-5dc27b7c=\\"\\" class=\\"text-field\\"><label data-v-5dc27b7c=\\"\\">Name</label>
                    <div data-v-5dc27b7c=\\"\\" class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field\\">
                      <!---->
                      <div class=\\"v-input__control\\">
                        <div class=\\"v-field v-field--active v-field--center-affix v-field--dirty v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr\\">
                          <div class=\\"v-field__overlay\\"></div>
                          <div class=\\"v-field__loader\\">
                            <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                              <!---->
                              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                                <div class=\\"v-progress-linear__indeterminate\\">
                                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                                </div>
                              </transition-stub>
                              <!---->
                            </div>
                          </div>
                          <!---->
                          <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                            <!----><label class=\\"v-label v-field-label\\" for=\\"input-1\\">
                              <!---->
                              <!---->
                            </label>
                            <!---->
                            <div class=\\"v-field__input\\" data-no-activator=\\"\\"><input name=\\"name\\" size=\\"1\\" type=\\"text\\" id=\\"input-1\\" aria-describedby=\\"input-1-messages\\"></div>
                            <!---->
                          </div>
                          <!---->
                          <!---->
                          <div class=\\"v-field__outline\\">
                            <!---->
                            <!---->
                          </div>
                        </div>
                      </div>
                      <!---->
                      <div class=\\"v-input__details\\">
                        <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-1-messages\\">
                          <!---->
                        </transition-group-stub>
                        <!---->
                      </div>
                    </div>
                  </div>
                  <div data-v-5dc27b7c=\\"\\" class=\\"text-field\\"><label data-v-5dc27b7c=\\"\\">Email</label>
                    <div data-v-5dc27b7c=\\"\\" class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field\\">
                      <!---->
                      <div class=\\"v-input__control\\">
                        <div class=\\"v-field v-field--active v-field--center-affix v-field--dirty v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr\\">
                          <div class=\\"v-field__overlay\\"></div>
                          <div class=\\"v-field__loader\\">
                            <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                              <!---->
                              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                                <div class=\\"v-progress-linear__indeterminate\\">
                                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                                </div>
                              </transition-stub>
                              <!---->
                            </div>
                          </div>
                          <!---->
                          <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                            <!----><label class=\\"v-label v-field-label\\" for=\\"input-3\\">
                              <!---->
                              <!---->
                            </label>
                            <!---->
                            <div class=\\"v-field__input\\" data-no-activator=\\"\\"><input name=\\"email\\" size=\\"1\\" type=\\"text\\" id=\\"input-3\\" aria-describedby=\\"input-3-messages\\"></div>
                            <!---->
                          </div>
                          <!---->
                          <!---->
                          <div class=\\"v-field__outline\\">
                            <!---->
                            <!---->
                          </div>
                        </div>
                      </div>
                      <!---->
                      <div class=\\"v-input__details\\">
                        <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-3-messages\\">
                          <!---->
                        </transition-group-stub>
                        <!---->
                      </div>
                    </div>
                  </div>
                  <div data-v-5dc27b7c=\\"\\" class=\\"text-field\\"><label data-v-5dc27b7c=\\"\\">Phone</label>
                    <div data-v-5dc27b7c=\\"\\" class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field\\">
                      <!---->
                      <div class=\\"v-input__control\\">
                        <div class=\\"v-field v-field--active v-field--center-affix v-field--dirty v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr\\">
                          <div class=\\"v-field__overlay\\"></div>
                          <div class=\\"v-field__loader\\">
                            <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                              <!---->
                              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                                <div class=\\"v-progress-linear__indeterminate\\">
                                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                                </div>
                              </transition-stub>
                              <!---->
                            </div>
                          </div>
                          <!---->
                          <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                            <!----><label class=\\"v-label v-field-label\\" for=\\"input-5\\">
                              <!---->
                              <!---->
                            </label>
                            <!---->
                            <div class=\\"v-field__input\\" data-no-activator=\\"\\"><input name=\\"phone\\" size=\\"1\\" type=\\"text\\" id=\\"input-5\\" aria-describedby=\\"input-5-messages\\"></div>
                            <!---->
                          </div>
                          <!---->
                          <!---->
                          <div class=\\"v-field__outline\\">
                            <!---->
                            <!---->
                          </div>
                        </div>
                      </div>
                      <!---->
                      <div class=\\"v-input__details\\">
                        <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-5-messages\\">
                          <!---->
                        </transition-group-stub>
                        <!---->
                      </div>
                    </div>
                  </div>
                  <div data-v-5dc27b7c=\\"\\" class=\\"text-field\\"><label data-v-5dc27b7c=\\"\\">Website</label>
                    <div data-v-5dc27b7c=\\"\\" class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field\\">
                      <!---->
                      <div class=\\"v-input__control\\">
                        <div class=\\"v-field v-field--active v-field--center-affix v-field--dirty v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr\\">
                          <div class=\\"v-field__overlay\\"></div>
                          <div class=\\"v-field__loader\\">
                            <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                              <!---->
                              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
                              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                                <div class=\\"v-progress-linear__indeterminate\\">
                                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                                </div>
                              </transition-stub>
                              <!---->
                            </div>
                          </div>
                          <!---->
                          <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                            <!----><label class=\\"v-label v-field-label\\" for=\\"input-7\\">
                              <!---->
                              <!---->
                            </label>
                            <!---->
                            <div class=\\"v-field__input\\" data-no-activator=\\"\\"><input name=\\"website\\" placeholder=\\"https://\\" size=\\"1\\" type=\\"text\\" id=\\"input-7\\" aria-describedby=\\"input-7-messages\\"></div>
                            <!---->
                          </div>
                          <!---->
                          <!---->
                          <div class=\\"v-field__outline\\">
                            <!---->
                            <!---->
                          </div>
                        </div>
                      </div>
                      <!---->
                      <div class=\\"v-input__details\\">
                        <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-7-messages\\">
                          <!---->
                        </transition-group-stub>
                        <!---->
                      </div>
                    </div>
                  </div><button data-v-3c9d0845=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--flat v-theme--light bg-transparent v-btn--density-default v-btn--size-default v-btn--variant-elevated\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> Cancel </span>
                    <!---->
                    <!---->
                  </button><button data-v-3c9d0845=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--flat v-theme--light bg-secondary v-btn--density-default v-btn--size-default v-btn--variant-elevated\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> Next </span>
                    <!---->
                    <!---->
                  </button>
                </div>
              </div>
            </transition-stub>
            <transition-stub appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" disabled=\\"true\\">
              <div class=\\"v-window-item\\" style=\\"display: none;\\">
                <!---->
              </div>
            </transition-stub>
            <!---->
          </div>
          <!---->
        </div>
      </form>
    </div>
  </div>
</div>"
`;
